core/new-gui/src/app/workspace/component/result-panel/result-panel.component.ts

```diff
@@ -1,15 +1,169 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { DataSource } from '@angular/cdk/table';
+import { MatPaginator, MatTableDataSource } from '@angular/material';
 
+
+import { ExecuteWorkflowService } from './../../service/execute-workflow/execute-workflow.service';
+
+import { Observable } from 'rxjs/Observable';
+import { NgbModal , ModalDismissReasons } from '@ng-bootstrap/ng-bootstrap';
+import { ExecutionResult, SuccessExecutionResult } from './../../types/workflow-execute.interface';
+
+/**
+ * ResultPanelCompoent is the bottom level area that
+ *  displays the execution result of a workflow after
+ *  the execution finishes.
+ *
+ * The Component will display the result in an excel
+ *  table format, where each row represents a result
+ *  from the workflow and each column represents the
+ *  type of result the workflow returns.
+ *
+ * Clicking each row of the result table will create an
+ *  pop-up window and display the detail of that row
+ *  in a pretty json format.
+ *
+ * @author Henry Chen
+ * @author Zuozhi Wang
+ *
+ */
 @Component({
   selector: 'texera-result-panel',
   templateUrl: './result-panel.component.html',
   styleUrls: ['./result-panel.component.scss']
 })
 export class ResultPanelComponent implements OnInit {
 
-  constructor() { }
+  public showMessage: boolean = false;
+  public message: string = '';
+
+  public currentColumns: TableColumn[] | undefined;
+  public currentDisplayColumns: string[] | undefined;
+  public currentDataSource: MatTableDataSource<object> | undefined;
+  public currentDisplayRow: string = '';
+
+  @ViewChild(MatPaginator) paginator: MatPaginator | null = null;
+
+  constructor(private executeWorkflowService: ExecuteWorkflowService, private modalService: NgbModal) {
+    // once an execution has ended, update the result panel to dispaly
+    //  execution result or error
+    this.executeWorkflowService.getExecuteEndedStream().subscribe(
+      executionResult => this.handleResultData(executionResult),
+    );
+  }
+
+  /**
+   * Opens the ng-bootstrap model to display the row details in
+   *  pretty json format when clicked. User can view the details
+   *  in a larger, expanded format.
+   *
+   * @param content
+   */
+  public open(content: any): void {
+    this.modalService.open(content).result.then((result) => {
+      console.log(result);
+    }, (reason) => {
+      console.log(reason);
+    });
+  }
+
+
+  /**
+   * Fetches the information from the selected row in the result panel,
+   *  updates the content of the row to dispaly, and opens the
+   *  result panel to display the details of the selected rows.
+   *
+   * @param row
+   * @param content
+   */
+  public getRowDetails(row: any, content: any): void {
```

Comment:
Didn't we disallow the use of 'any'?

Reply:
Same comment, we do not allow `any` unless there's absolute no workaround. This is a strict rule. Please change this function and the function `open` above with property type signature.

Reply:
I have edited the type for `content` to `TemplateRef<any>` as described in this article,  https://blog.angular-university.io/angular-ng-template-ng-container-ngtemplateoutlet/.

As for the `row`, it represents each of the result instance fetched from backend, which have no defined format. Therefore, I kept it as type `any`.